src/ImageSharp/Memory/Buffer2D{T}.cs

```diff
@@ -62,17 +71,76 @@ internal Buffer2D(MemorySource<T> memorySource, int width, int height)
                 DebugGuard.MustBeLessThan(x, this.Width, nameof(x));
                 DebugGuard.MustBeLessThan(y, this.Height, nameof(y));
 
-                Span<T> span = this.GetSpan();
-                return ref span[(this.Width * y) + x];
+                return ref this.GetRowSpanUnchecked(y)[x];
             }
         }
 
+        /// <summary>
+        /// Gets a <see cref="Span{T}"/> to the row 'y' beginning from the pixel at the first pixel on that row.
+        /// </summary>
+        /// <param name="y">The y (row) coordinate.</param>
+        /// <returns>The <see cref="Span{T}"/>.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Span<T> GetRowSpan(int y)
+        {
+            Guard.MustBeGreaterThanOrEqualTo(y, 0, nameof(y));
+            Guard.MustBeLessThan(y, this.Height, nameof(y));
+
+            return this.cachedMemory.Length > 0
+                ? this.cachedMemory.Span.Slice(y * this.Width, this.Width)
+                : this.GetRowMemorySlow(y).Span;
+        }
+
         /// <summary>
         /// Disposes the <see cref="Buffer2D{T}"/> instance
         /// </summary>
         public void Dispose()
         {
-            this.MemorySource.Dispose();
+            this.MemoryGroup.Dispose();
+            this.cachedMemory = default;
+        }
+
+        /// <summary>
+        /// Same as <see cref="GetRowSpan"/>, but does not validate index in Release mode.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal Span<T> GetRowSpanUnchecked(int y)
+        {
+            DebugGuard.MustBeGreaterThanOrEqualTo(y, 0, nameof(y));
+            DebugGuard.MustBeLessThan(y, this.Height, nameof(y));
+
+            return this.cachedMemory.Length > 0
+                ? this.cachedMemory.Span.Slice(y * this.Width, this.Width)
+                : this.GetRowMemorySlow(y).Span;
+        }
+
+        /// <summary>
+        /// Gets a <see cref="Memory{T}"/> to the row 'y' beginning from the pixel at the first pixel on that row.
+        /// This method is intended for internal use only, since it does not use the indirection provided by
+        /// <see cref="MemoryGroupView{T}"/>.
+        /// </summary>
+        /// <param name="y">The y (row) coordinate.</param>
+        /// <returns>The <see cref="Span{T}"/>.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
More of a question than anything. When should we use `InliningOptions`?

Reply:
Always! Forgot it by mistake :smile: